libs/network/socketserver.c

```diff
@@ -534,20 +579,35 @@ bool socketClientConnectionsIdle(JsNetwork *net) {
             if (num > 0) {
               if (!receiveData) {
                 receiveData = jsvNewFromEmptyString();
-                jsvObjectSetChild(connection, HTTP_NAME_RECEIVE_DATA, receiveData);
               }
               if (receiveData) { // could be out of memory
                 jsvAppendStringBuf(receiveData, buf, (size_t)num);
-                if ((socketType&ST_TYPE_MASK)==ST_HTTP && !hadHeaders) {
-                  // for HTTP see whether we now have full response headers
-                  JsVar *resVar = jsvObjectGetChild(connection,HTTP_NAME_RESPONSE_VAR,0);
-                  if (httpParseHeaders(&receiveData, resVar, false)) {
-                    hadHeaders = true;
-                    jsvObjectSetChildAndUnLock(connection, HTTP_NAME_HAD_HEADERS, jsvNewFromBool(hadHeaders));
-                    jsiQueueObjectCallbacks(connection, HTTP_NAME_ON_CONNECT, &resVar, 1);
+
+                if ((socketType&ST_TYPE_MASK)==ST_UDP) {
+                  JsVar *address = jsvNewFromEmptyString(); // inet_ntoa replacement
```

Comment:
The client/server code was pretty badly duplicated before - did you come across any reason why we couldn't un-duplicate it? It's probably a bit too much to add to the PR, but I could file an issue for it afterwards?